src/plugins/dashboard/public/services/dashboard_content_management/lib/check_for_duplicate_dashboard_title.ts

```diff
@@ -44,21 +50,62 @@ export async function checkForDuplicateDashboardTitle(
     return true;
   }
 
-  const { hits } = await contentManagement.client.search<
-    DashboardCrudTypes['SearchIn'],
-    DashboardCrudTypes['SearchOut']
-  >({
-    contentTypeId: DASHBOARD_CONTENT_ID,
-    query: {
-      text: title ? `${title}*` : undefined,
-      limit: 10,
-    },
-    options: { onlyTitle: true },
-  });
-  const duplicate = hits.find((hit) => hit.attributes.title.toLowerCase() === title.toLowerCase());
+  const [baseDashboardName, duplicationId] = extractTitleAndCount(title);
+
+  let searchMatchPageHits;
+  let searchMatchPaginationObject;
+
+  ({ hits: searchMatchPageHits, pagination: searchMatchPaginationObject } =
+    await contentManagement.client.search<
+      DashboardCrudTypes['SearchIn'],
+      DashboardCrudTypes['SearchOut']
+    >({
+      contentTypeId: DASHBOARD_CONTENT_ID,
+      query: {
+        text: title ? `${baseDashboardName}*` : undefined,
+        limit: searchLimit,
+        cursor: String(duplicationId ? Math.ceil(duplicationId / searchLimit) : 1),
+      },
+      options: { onlyTitle: true },
+    }));
+
+  const isMatchedResultInvalid =
+    // Given we are looking to find the collection of dashboards that's closest to the duplicated dashboard with the last incremented number,
+    // we compare the last item in the page result we got against the total hits to determine if we should self correct
+    searchMatchPageHits.length &&
+    extractTitleAndCount(searchMatchPageHits.slice(-1)[0].attributes.title)[1] >
+      searchMatchPaginationObject.total;
+
+  if (isMatchedResultInvalid) {
+    ({ hits: searchMatchPageHits, pagination: searchMatchPaginationObject } =
+      await contentManagement.client.search<
```

Comment:
Why exactly do we need to run the search again? Or use a cursor? It seems to me this could work quite well with **one** search of ~20 or so Dashboards with no cursor where we simply loop through the results to find the highest number and increment that.

If we miss and accidentally return a number that isn't unique, nothing really breaks, the user just has to increment it themselves.

Reply:
Before switching the previous implementation, I'd used a cursor so we are able to fetch the last page of the search so we can pick the last created dashboard without having to loop multiple times. I opted for this approach because if we have a situation where the duplicated dashboard count exceeds search limit (right now it's 10) the returned page would not give us a way to determine the highest number without performing multiple loops.  

I have switched the implementation though now so we are able to sort and simply pick the the last incremented dashboard.

Reply:
I think that's okay. If the user has more than 10 Dashboards with the same basic title structure, it's completely okay if this doesn't recommend a unique title in ~5% or less of cases. Remember that the behaviour we're replacing simply _told_ the user their title wasn't unique and let them figure out how to make it unique.